web-common/src/features/dashboards/stores/dashboard-stores.ts

```diff
@@ -182,8 +182,10 @@ const metricViewReducers = {
         metricsView,
         fullTimeRange,
       );
+      mergeDashboardWithLocalPreferences(state.entities[name]);
```

Comment:
True. But since this is updating the state it will be performant if we didnt replace the entire dashboard state. Will be especially impactful if and when we use immer.